byway/src/main/java/com/google/sps/servlets/StopsServlet.java

```diff
@@ -14,35 +14,66 @@
 
 package com.google.sps.servlets;
 
-import com.google.errorprone.annotations.concurrent.GuardedBy;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
 import com.google.gson.Gson;
+import com.google.sps.data.Trip;
+import java.lang.reflect.Type;
+import com.google.gson.reflect.TypeToken;
 import java.io.IOException;
 import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.servlet.annotation.MultipartConfig;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
+
 /** Servlet that handles data for stops */
 @WebServlet("/api/stop")
 public final class StopsServlet extends HttpServlet {
-  @GuardedBy("this")
-  private ArrayList<String> stops= new ArrayList<String>();
   private final Gson gson = new Gson(); 
-  // Implement datastore in routepage branch
+  private final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+  private static final Type ARRAYLIST_STRING = new TypeToken<ArrayList<String>>() {}.getType();
 
+  /* Passes saved destinations stops (if any) to be shown in the schedule panel */
   @Override
-  public synchronized void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Gson gson = new Gson();
-    String json = gson.toJson(stops);
-    
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    List<String> stops= new ArrayList<>();
```

Comment:
If you want a default empty list which you won't need to be able to append to, you can use `Collections.emptyList()`, rather than constructing a new empty `ArrayList`. This avoids allocation, because there is only one instance of `Collections.emptyList()` for the entire program. (If you do need to `add` or `remove` anything, that won't work, but if you just want to put an empty list in a variable, and aren't going to `add` or `remove` (replacing the thing with a different list is fine of course), you can just use `emptyList()`

Reply:
I think I'll use emptyList since I am replacing the list with the destinations list in datastore. Thanks for explaining.